src/pydata_sphinx_theme/assets/styles/components/_search.scss

```diff
@@ -75,11 +73,16 @@
   align-items: center;
   align-content: center;
   color: var(--pst-color-text-muted);
-  // Needed to match other icons hover
-  padding: 0 0 0.25rem 0;
   border-radius: 0;
+  border: none; // Override Bootstrap button border
+  font-size: 1rem; // Override Bootstrap button font size
```

Comment:
For both the search button and the theme switcher button, I decided to go ahead and make them consistent with the other icons at the end of the nav bar.